DTAdmin/GroupVC.swift

```diff
@@ -0,0 +1,117 @@
+            if responce.statusCode == 200 {
+                groups = result
+                HTTPService.getAllData(entityName: "faculty"){ (result:[Faculty],responce) in
+                    if responce.statusCode == 200 {
+                        faculties = result
+                        HTTPService.getAllData(entityName: "speciality"){ (result:[Speciality],responce) in
+                            if responce.statusCode == 200 {
+                                specialities = result
+                                var commonData:[[String:String]] = []
+                                for group in groups {
+                                    var newGroup:[String:String] = [:]
+                                    newGroup["group_name"] = group.group_name
+                                    newGroup["group_id"] = group.group_id
+                                    newGroup["faculty_id"] = group.faculty_id
+                                    newGroup["speciality_id"] = group.speciality_id
+                                    for faculty in faculties {
+                                        if (group.faculty_id == faculty.faculty_id){
+                                            newGroup["faculty_name"] = faculty.faculty_name
+                                            newGroup["faculty_description"] = faculty.faculty_description
+                                        }
+                                    }
+                                    for speciality in specialities {
+                                        if (group.speciality_id == speciality.speciality_id){
+                                            newGroup["speciality_code"] = speciality.speciality_code
+                                            newGroup["speciality_name"] = speciality.speciality_name
+                                        }
+                                    }
+                                    commonData.append(newGroup)
+                                }
+                                completion(commonData)
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    
+    /*
+     // MARK: - Navigation
+     
+     // In a storyboard-based application, you will often want to do a little preparation before navigation
+     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+     // Get the new view controller using segue.destinationViewController.
+     // Pass the selected object to the new view controller.
+     }
+     */
+    
+}
+
+//MARK: table view delegate
+extension GroupVC : UITableViewDelegate {
+   
+}
+
+//MARK: table view data source
+extension GroupVC : UITableViewDataSource {
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = groupTableView.dequeueReusableCell(withIdentifier: "GroupCell")! as UITableViewCell
```

Comment:
Do not use force unwrap. Use `func dequeueReusableCell(withIdentifier identifier: String, for indexPath: IndexPath) -> UITableViewCell` which returns non-optional cell